### PR TITLE
Quadtree

### PR DESCRIPTION
**Pull request by Amyantis**
Quadtree:
- split
- unsplit
- get nodes at position
- draw zones and debug node shape (with shortcut)

### CODE REVIEW

File: engine/src/QuadTree.h [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    void operator=(const QuadTree&) = delete;
+public:
+    QuadTree(const unsigned int &x = 0,
+             const unsigned int &y = 0,
+             const unsigned int &width = 1,
+             const unsigned int &height = 1,
+             const unsigned int &node_capacity_ = 5,
+             const unsigned int &level_ = 0);
+
+    void setNode_capacity(unsigned int n);
```
Comment (terahlunah) [inline engine/src/QuadTree.h]: For consistency, you should name the method "setNodeCapacity"

Comment (Amyantis) [inline engine/src/QuadTree.h]: You're right. Thank you.

File: engine/src/QuadTree.h [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    void operator=(const QuadTree&) = delete;
+public:
+    QuadTree(const unsigned int &x = 0,
+             const unsigned int &y = 0,
+             const unsigned int &width = 1,
+             const unsigned int &height = 1,
+             const unsigned int &node_capacity_ = 5,
+             const unsigned int &level_ = 0);
+
+    void set_NodeCapacity(unsigned int n);
```
Comment (terahlunah) [inline engine/src/QuadTree.h]: Why do you put an underscore in the name ?

Comment (Amyantis) [inline engine/src/QuadTree.h]: To deeply prevent the ambiguity you underlined, why ? (the attribute is now named NodeCapacity)

Comment (terahlunah) [inline engine/src/QuadTree.h]: If you look at other functions like "setShape" there is no underscore in it (it's not written set_Shape)
So why do you want to have one in set_NodeCapacity ? Just use setNodeCapacity (without underscore)

Comment (Amyantis) [inline engine/src/QuadTree.h]: It was to avoid the reading "setNode Capacity", but I agree this is not a good idea.

**Pull request merged**